### PR TITLE
Provide user facing options to control if expression bodied members are preferred or not.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide Please review at low priority.

**Comment by CyrusNajmabadi** [comment]
retest windows_debug_unit64_prtest  please

**Comment by CyrusNajmabadi** [comment]
Closed side is here: https://github.com/dotnet/roslyn-internal/pull/1314

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-ide  plz review when convenient.

**Comment by CyrusNajmabadi** [comment]
retest windows_release_unit64_prtest  please

File: src/Workspaces/CSharp/Portable/Extensions/TypeSyntaxExtensions.cs [hunk 1]
```diff
@@ -12,6 +12,12 @@ namespace Microsoft.CodeAnalysis.CSharp.Extensions
 {
     internal static class TypeSyntaxExtensions
     {
+        public static bool IsVoid(this TypeSyntax typeSyntax)
+        {
+            return typeSyntax.IsKind(SyntaxKind.PredefinedType) &&
+                ((PredefinedTypeSyntax)typeSyntax).Keyword.IsKind(SyntaxKind.VoidKeyword);
+        }
```
Comment (DustinCampbell) [inline src/Workspaces/CSharp/Portable/Extensions/TypeSyntaxExtensions.cs]: Really? We don't have this helper already?

File: src/EditorFeatures/CSharpTest/UseExpressionBody/UseExpressionBodyForPropertiesTests.cs [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+    {
+        get
+        {
+            throw new NotImplementedException(); // comment
+        }
+    }
+}", compareTokens: false, options: UseBlockBody);
+        }
+    }
+}
```
Comment (DustinCampbell) [inline src/EditorFeatures/CSharpTest/UseExpressionBody/UseExpressionBodyForPropertiesTests.cs]: Need a test to not be available when an accessor has attributes.

``` C#
int P {
    [A] get { return 42; }
}
```

**Comment by CyrusNajmabadi** [comment]
@DustinCampbell I added support for comments and i disabled the fix for getters with attributes on them.

**Comment by CyrusNajmabadi** [comment]
retest windows_release_unit64_prtest  please

**Pull request merged**